src/v/kafka/requests/fetch_request.cc

```diff
@@ -340,6 +365,16 @@ read_from_ntp(op_context& octx, model::ntp ntp, fetch_config config) {
               return make_ready_partition_response_error(
                 error_code::not_leader_for_partition);
           }
+          if (mntpv.is_materialized()) {
+              if (auto log = mgr.log(mntpv.input_ntp())) {
```

Comment:
use the pattern `if (auto c = ...; c) {}` available in c++17. but no need to fix this in the current pr.

Reply:
oh hot i didn't know about that feature